cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_train_dense_default_impl.i

```diff
@@ -117,26 +117,26 @@ public:
     bool terminateCriteria(ImpurityData & imp, algorithmFPType impurityThreshold, size_t nSamples) const { return imp.value() < impurityThreshold; }
 
     template <typename BinIndexType>
-    int findBestSplitForFeatureSorted(algorithmFPType * featureBuf, IndexType iFeature, const IndexType * aIdx, size_t n, size_t nMinSplitPart,
-                                      const ImpurityData & curImpurity, TSplitData & split, const algorithmFPType minWeightLeaf,
-                                      const algorithmFPType totalWeights, const BinIndexType * binIndex) const;
+    int64_t findBestSplitForFeatureSorted(algorithmFPType * featureBuf, IndexType iFeature, const IndexType * aIdx, size_t n, size_t nMinSplitPart,
```

Comment:
What is the need to change this type from 32 to 64 bytes?

Reply:
As @KulikovNikita said, using vanilla `int` is banned from the library. `IndexType` as of today can be `uint8_t`, `uint16_t`, `int` (for `int` see `typedef int IndexType`). If we change `int` to `uint32_t` as the `TODO` requests, `int32_t` might overflow. Note that this is only used for local variables on the stack, no we're not talking about big chunks of memory here.

Reply:
`int` is not banned in DAAL kernels. Otherwise, most part of theirs code is banned. I guess there should be separate discussion for DAAL

Reply:
There is a defined IndexType used within RF which probably should be used. Unfortunately the code uses it inconsistently.

Reply:
To be honest, the whole reason why its returning an int is poorly implemented, as it is used as a quasi-flag to trigger finalizeBestSplit. Why this roundabout logic was used is beyond me.  I think we should have a talk about to what level RF needs to be refactored vs just repaired. I think some of the issues such as the constFeatures is a symptom of too much tech debt.

Reply:
Reverted it, but I don't think I agree with keeping the return type `int`. At the very least, we're using `size_t`, `IndexType`, and raw integer types inconsistently, and this discussion didn't explain why. If that should be refactored separately, fine.